todo_app/data/session_items.py

```diff
@@ -67,3 +67,22 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+
+def remove_item(item):
+    """
+    Removes an existing item from the session. If no existing item matches the ID of the specified item, nothing is removed.
+
+    Args:
+        item: The item to remove.
+    """
+    existing_items = get_items()
+
+    for i in range(len(existing_items)):
+        if existing_items[i] is item:
+            del existing_items[i]
+            break
```

Comment:
Not a recommendation but worth seeing as a comparison, how we might write that with a list comprehension:
```python
    existing_items = get_items()

    new_items = [existing_item for existing_item in existing_items if existing_item != item]

    session['items'] = new_items
```


For both methods we could also equally do this just by comparing item ids rather than items fully, so we might choose to change the signature of this method to just take the id rather than requiring an item look up

Reply:
Thanks for the prompt!